registry/repository.go

```diff
@@ -32,35 +37,68 @@ type RepositoryOptions struct {
 
 // repositoryClient implements Repository
 type repositoryClient struct {
-	registry.Repository
+	// oras.Target specifies the type of the target.
+	// Implementations that are supported in Notation:
+	// remote.Repository (https://pkg.go.dev/oras.land/oras-go/v2@v2.0.1/registry/remote#Repository)
+	// oci.Store (https://pkg.go.dev/oras.land/oras-go/v2@v2.0.1/content/oci#Store)
+	oras.Target
```

Comment:
Should this be a `GraphTarget` so `ListSignatures` can implemented upon it?

Reply:
BTW this is a different option than `graphTargetClient` so you don't need to do the type assertion for `Resolve`, `ListSignatures` and `FetchSignatureBlob`.

Reply:
We need this also to be a `Pusher` as it's calling `oras.Pack()` to push signatures.